x-pack/legacy/plugins/lens/public/editor_frame_plugin/embeddable/embeddable.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+
+import { TimeRange } from 'ui/timefilter/time_history';
+import { Query, StaticIndexPattern, ExpressionRenderer } from 'src/legacy/core_plugins/data/public';
+import { Filter } from '@kbn/es-query';
+import { Subscription } from 'rxjs';
+import {
+  Embeddable as AbstractEmbeddable,
+  EmbeddableOutput,
+  IContainer,
+  EmbeddableInput,
+} from '../../../../../../../src/legacy/core_plugins/embeddable_api/public/index';
+import { Document, DOC_TYPE } from '../../persistence';
+import { ExpressionWrapper } from './expression_wrapper';
+
+export interface LensEmbeddableConfiguration {
+  savedVis: Document;
+  editUrl: string;
+  editable: boolean;
+  indexPatterns?: StaticIndexPattern[];
+}
+
+export interface LensEmbeddableInput extends EmbeddableInput {
+  timeRange?: TimeRange;
+  query?: Query;
+  filters?: Filter[];
+}
+
+export interface LensEmbeddableOutput extends EmbeddableOutput {
+  indexPatterns?: StaticIndexPattern[];
+}
+
+export class Embeddable extends AbstractEmbeddable<LensEmbeddableInput, LensEmbeddableOutput> {
+  type = DOC_TYPE;
+
+  private expressionRenderer: ExpressionRenderer;
+  private savedVis: Document;
+  private domNode: HTMLElement | Element | undefined;
+  private subscription: Subscription;
+
+  private currentContext: {
+    timeRange?: TimeRange;
+    query?: Query;
+    filters?: Filter[];
+    lastReloadRequestTime?: number;
+  } = {};
+
+  constructor(
+    expressionRenderer: ExpressionRenderer,
+    { savedVis, editUrl, editable, indexPatterns }: LensEmbeddableConfiguration,
+    initialInput: LensEmbeddableInput,
+    parent?: IContainer
+  ) {
+    super(
+      initialInput,
+      {
+        defaultTitle: savedVis.title,
+        savedObjectId: savedVis.id!,
+        editable,
+        editUrl,
+        indexPatterns,
```

Comment:
Okay I've looked into this more and I think the only thing that I would like you to change is adding a comment indicating that this is used by the dashboard embeddable